acinclude/squid-util.m4

```diff
@@ -214,6 +214,31 @@ AC_DEFUN([SQUID_YESNO],[
   AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([Bad argument for $2: "$1". Expecting "yes", "no", or no argument.])])
 ])
 
+dnl Check that a library is actually available, useable,
+dnl and where its pieces are (eg headers and hack macros)
+dnl Parameters for this macro are:
+dnl 1) library name (without 'lib' prefix)
+dnl 2) necessary library checks (to be executed by this macro unless the use of the library is disabled)
+dnl   These checks should set LIBFOO_LIBS automake variable (on success)
+dnl   and ensure that it is empty or unset (on failures).
+AC_DEFUN([SQUID_CHECK_LIB_WORKS],[
+AS_IF([m4_translit([test "x$with_$1" != "xno"], [-+.], [___])],[
+  $2
+  AS_IF([! test -z m4_toupper(m4_translit(["$LIB$1_LIBS"], [-+.], [___]))],[
+    m4_toupper(m4_translit([CPPFLAGS="$LIB$1_CFLAGS $CPPFLAGS"], [-+.], [___]))
+    m4_toupper(m4_translit([LIB$1_LIBS="$LIB$1_PATH $LIB$1_LIBS"], [-+.], [___]))
+    AC_MSG_NOTICE([Library '$1' support: m4_translit([${with_$1:=yes (auto)} m4_toupper($LIB$1_LIBS)], [-+.], [___])])
+    m4_translit([with_$1], [-+.], [___])=yes
```

Comment:
I am concerned that this successful code path condition (i.e. `test -z...`) is not identical to the successful code path in some `$2` code: The caller and this code are using slightly different tests to determine what "success" means. I do _not_ claim that those tests  already result in different outcomes today.

I wonder whether we should split existing `$2` argument into two:

* `$2`: Required code that evaluates things and sets LIBFOO_LIBS upon success.
* `$3`: Optional code that reacts to successful outcome of the above code. That `$3` code would be executed somewhere inside this `if ! test -z` statement, probably after the above assignment.

The split will ensure that there is only one "success" condition -- the one computed by the above `test -z...` code.

I do not insist on this change.

Reply:
This is one of the things which are being left out of scope to reduce the amount of change in this PR. It will happen in a followup once this is merged.

Reply:
FTR: as far as I can tell the difference between the old `test "x$foo" != "x"` and the new `! test -z ...` is that the new condition has an explicit state difference for the `$foo` being unset vs set-to-empty. For our uses these are equivalent, allowing the slightly more optimal `test -z ...` to be used.